library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -615,23 +620,30 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
         @ScaledPan val value = if (horizontal) scaledPanX else scaledPanY
         val viewSize = if (horizontal) mContainerWidth else mContainerHeight
         @ScaledPan val contentSize = if (horizontal) mTransformedRect.width() else mTransformedRect.height()
+        @ScaledPan val basePanValue = computeBasePan()[(if (horizontal) 0 else 1)]
 
         val overScrollable = if (horizontal) mOverScrollHorizontal else mOverScrollVertical
         @ScaledPan val overScroll = (if (overScrollable && allowOverScroll) maxOverScroll else 0).toFloat()
-        return getPanCorrection(value, viewSize, contentSize, overScroll)
+        return getPanCorrection(value, viewSize, contentSize, overScroll, basePanValue)
     }
 
     @ScaledPan
     private fun getPanCorrection(@ScaledPan value: Float, viewSize: Float,
-                                 @ScaledPan contentSize: Float, @ScaledPan overScroll: Float): Float {
+                                 @ScaledPan contentSize: Float, @ScaledPan overScroll: Float,
+                                 @ScaledPan basePanValue: Float): Float {
         @ScaledPan val tolerance = overScroll.toInt()
         var min: Float
         var max: Float
         if (contentSize <= viewSize) {
-            // If contentSize <= viewSize, we want to stay centered.
-            // Need a positive translation, that shows some background.
-            min = (viewSize - contentSize) / 2f
-            max = (viewSize - contentSize) / 2f
+            if(mSmallerPolicy == ZoomApi.SMALLER_POLICY_FROM_TRANSFORMATION) {
```

Comment:
Using a `when` statement for this kind of check is much more elegant imo.

sth. like:
```
// contentSize <= viewSize, use mSmallerPolicy to decide how to correct it
when (mSmallerPolicy) {
    ZoomApi.SMALLER_POLICY_FROM_TRANSFORMATION -> {
        min = basePanValue
        max = basePanValue
    }
    else -> {
        // we want to stay centered.
        // Need a positive translation, that shows some background.
        min = (viewSize - contentSize) / 2f
        max = (viewSize - contentSize) / 2f
    }
}
```